favorite_files.py

```diff
@@ -112,13 +112,17 @@ def add(self, names, group_name=None):
         added = 0
         # Iterate names and add them to group/global if not already added
         for n in names:
-            if not Favs.exists(n, group_name=group_name):
+            if Favs.file_index(n, group_name=group_name) is None:
                 if exists(n):
                     Favs.set(n, group_name=group_name)
                     added += 1
                 else:
                     # File does not exist on disk; cannot add
                     disk_omit_count += 1
+            else:
+                message = "is already among your favorite files and has not been added."
+                print(n, message)   # full filename printed to the console
```

Comment:
Please use `error` which will use notifications or the Sublime error dialog for this.

If we want to also log something in the console, use something like this which will make it clear where the log came from:

```py
def log(msg):
    """Log message."""

    print("FavoriteFiles: %s" % msg)
```